classes/form/CustomerPersister.php

```diff
@@ -188,6 +188,19 @@ private function create(Customer $customer, $clearTextPassword)
             return false;
         }
 
+        /**
+         * Database optimization
+         * We check if the email has been assigned to guest account
+         */
+        $check = Customer::customerExists($customer->email, true, false);
```

Comment:
Let's use a explicit name for the data stored in it, for instance `$customerId`

```suggestion
        $customerId = Customer::customerExists($customer->email, true, false);
```